### ISSUE TITLE
Synchronize with the upcoming Oxygen version 17.1 release

### ISSUE THREAD

**Issue by AlexJitianu** [TEIC/oxygen-tei#5]
Oxygen version 17.1 is not far away so we want to make sure the TEI framework contains the new changes and that the Java extensions are up to date. I will probably create a branch and make all these changes on it and when v17.1 is released I will merge it into the trunk.

**Comment by martindholmes** [comment]
Hi Alex. I haven't yet had a chance to update the build file so it will work on Windows. You can still build in the way you've always done (I'm not sure how you do that on Windows, though, since it's a bash script). The most important thing is to use the latest release version of P5 and the latest release version of the Stylesheets, which are currently here:

http://sourceforge.net/projects/tei/files/TEI-P5-all/tei-2.8.0.zip/download
http://sourceforge.net/projects/tei/files/Stylesheets/tei-xsl-7.39.0.zip/download

Cheers,
Martin

**Comment by AlexJitianu** [comment]
Hi Martin,

Thank you for the hints! Indeed, I remember I had to run that script on a Linux. I'll either try that again or use the direct download links you've provided.

**Comment by jamescummings** [comment]
What is the date of oXygen 17.1 release?  I'm assuming this couldn't be delayed until the next upcoming TEI release?

**Comment by martindholmes** [comment]
Hi Alex,

If you're running it on Linux anyway, you could try my ant task, which has a build target for you:

ant syncro

I'm presuming that after you've built the package, you sign the results for use in Oxygen -- is that right? 

Cheers,
Martin

**Comment by AlexJitianu** [comment]
Hi James,

The release is scheduled for mid October. We are doing the TEI update a 
little before that time because we want to make sure it is stable. When 
it's the next TEI release scheduled? I don't think we can delay the 
Oxygen release.

Best regards,
Alex
-- 
Alex Jitianu
<oXygen/>  XML Editor, Schema Editor and XSLT Editor/Debugger
http://www.oxygenxml.com

**Comment by AlexJitianu** [comment]
Hi Martin,
The built package will be bundled into Oxygen v17.1 (we have a number of predefined frameworks and TEI is one of them).

**Comment by martindholmes** [comment]
Hi Alex,
I don't think you need to wait for a TEI release, even if that were practical. We're still working out some of the issues around moving to GitHub. In my last question, I was actually asking about signing the code. If you don't do that, and all you actually do is build the package without any changes, then you could take the latest build from our build server (http://teijenkins.hcmc.uvic.ca/job/oxygen-tei-stable/).

**Comment by martindholmes** [comment]
Hi Alex,

We are planning the TEI release for October 8. The options for Oxygen are:

1. Build the plugin before October 8, so that you use the current stable versions linked above.

2. Build the plugin a few days after our release, so that we're all confident that the release has no problems.

We have transitioned from SourceForge to GitHub recently, and a lot of changes have been made to our release scripts and processes. We'd like to avoid a situation where you grab a brand-new release of P5 and the Stylesheets for Oxygen 17.1, and then we discover that we screwed something up in our release process and have to scramble to do a bugfix release. How would you prefer to proceed?

Cheers,
Martin

**Comment by AlexJitianu** [comment]
Hi Martin,

I've went with option number 1. I've already integrated the current, stable version. On the branch I've created I also brought in the small changes that we've made to the TEI framework and Java sources in the current Oxygen development stream. One aspect that should be taken into account is that this framework might not be fully compatible with older versions of Oxygen. I'm expecting the TEI community to work with different versions of Oxygen so we should probably tag or branch the current project state.

We'll see if we have time to integrate the release from October 8. If it doesn't make it perhaps we'll put it in a minor release afterwards.

Alex
